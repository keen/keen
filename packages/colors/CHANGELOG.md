# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [1.1.1](https://github.com/keen/keen/compare/@keen.io/colors@1.1.0...@keen.io/colors@1.1.1) (2022-05-12)

**Note:** Version bump only for package @keen.io/colors





# [1.1.0](https://github.com/keen/keen/compare/@keen.io/colors@1.0.0...@keen.io/colors@1.1.0) (2020-09-14)


### Features

* ðŸŽ¸ update readmes ([b2f70be](https://github.com/keen/keen/commit/b2f70bec7c7e73c4fd8012e7ce3b847d316e71a8))





# [1.0.0](https://github.com/keen/keen/compare/@keen.io/colors@0.2.0...@keen.io/colors@1.0.0) (2020-09-14)


### Features

* ðŸŽ¸ readme updates ([530c846](https://github.com/keen/keen/commit/530c846a7424c10260c08a3ee908252e0e39fecb))





# [0.2.0](https://github.com/keen/keen/compare/@keen.io/colors@0.1.2...@keen.io/colors@0.2.0) (2020-05-26)


### Features

* ðŸŽ¸ buttons ([33c14d6](https://github.com/keen/keen/commit/33c14d68d53ef9374ee8f28ec7d50dbf0f8c98e9))





## [0.1.2](https://github.com/keen/keen/compare/@keen.io/colors@0.1.1...@keen.io/colors@0.1.2) (2020-04-27)

**Note:** Version bump only for package @keen.io/colors





## [0.1.1](https://github.com/keen/keen/compare/@keen.io/colors@0.1.0...@keen.io/colors@0.1.1) (2020-04-15)

**Note:** Version bump only for package @keen.io/colors





# 0.1.0 (2020-03-16)


### Features

* ðŸŽ¸ legend ([386f140](https://github.com/keen/keen/commit/386f140b9c1626807988d152470708fed6075c56))
* ðŸŽ¸ pie chart ([e1063a9](https://github.com/keen/keen/commit/e1063a9db7c8a477b9bcc85f693442eece0c410d))
* ðŸŽ¸ responsive wrapper ([7bd729a](https://github.com/keen/keen/commit/7bd729afddb3df58e67e36c22a96a9bee497f803))
