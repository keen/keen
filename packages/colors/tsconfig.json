{
  "extends": "../../tsconfig.base.json",
  "compilerOptions": {
    "rootDir": "./src",
    "outDir": "./dist",
    "declarationDir": "./typings",
    "resolveJsonModule": true
  },
  "references": [{ "path": "../dev-tools" }],
  "include": ["./src/**/*"]
}
