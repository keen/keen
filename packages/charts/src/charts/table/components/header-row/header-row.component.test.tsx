import React from 'react';
import { render as rtlRender, fireEvent } from '@testing-library/react';
import { colors } from '@keen.io/colors';
import 'jest-styled-components';

import HeaderRow from './header-row.component';

const render = (overProps: any = {}) => {
  const props = {
    ...overProps,
    data: [
      { key: 'id', value: 'id' },
      { key: 'title', value: 'title' },
    ],
    color: colors.blue[300],
    sortOptions: null,
    onSort: jest.fn(),
    typography: {
      fontSize: 20,
      fontWeight: 'normal',
    },
    isColumnDragged: false,
  };

  const wrapper = rtlRender(
    <table>
      <HeaderRow {...props} />
    </table>
  );

  return {
    props,
    wrapper,
  };
};

test('renders columns according to provided data', () => {
  const {
    wrapper: { getByText },
  } = render();

  expect(getByText('id')).toBeInTheDocument();
  expect(getByText('title')).toBeInTheDocument();
});

test('allows user to sort data', () => {
  const {
    wrapper: { getByText },
    props,
  } = render();

  const element = getByText('id');
  fireEvent.click(element);

  expect(props.onSort).toHaveBeenCalledWith({
    propertyName: 'id',
    sortMode: 'ascending',
  });
});

test('renames columns based on configuration', () => {
  const columnsNamesMapping = {
    title: 'Product name',
  };

  const {
    wrapper: { getByText },
  } = render({ columnsNamesMapping });

  const element = getByText('Product name');

  expect(element).toBeInTheDocument();
});

test('applies typography settings to table header', () => {
  const {
    wrapper: { getByTestId },
  } = render();
  const headerElement = getByTestId('header-row-container');

  expect(headerElement).toMatchSnapshot();
});
