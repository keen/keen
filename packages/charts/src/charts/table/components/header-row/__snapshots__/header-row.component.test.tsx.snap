// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`applies typography settings to table header 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  text-align: left;
  background: inherit;
  background: #44748C;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 5px;
  box-sizing: border-box;
  min-height: 20px;
  padding: 20px;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: no-wrap;
  -ms-flex-wrap: no-wrap;
  flex-wrap: no-wrap;
  position: relative;
}

.c5 {
  position: relative;
  top: 5px;
  left: -1px;
}

.c0 {
  font-size: 20px;
  font-weight: normal;
}

.c1 {
  padding: 0;
  position: -webkit-sticky;
  position: sticky;
  top: 0;
  z-index: 1;
  background: #44748C;
}

<tr
  class="c0"
  data-testid="header-row-container"
>
  <td
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        id
        <div
          style="visibility: hidden;"
        >
          <div
            class="c4"
            data-testid="sort-indicators"
          >
            <div
              style="opacity: 0;"
            >
              <svg
                height="15"
                opacity="1"
                viewBox="0 0 24 30"
                width="10"
              >
                <polygon
                  fill="#FFFFFF"
                  fill-rule="evenodd"
                  points="10.768 0 10.768 23.16 3.245 15.637 .776 18.106 12.414 29.744 24.052 18.106 21.583 15.637 14.06 23.16 14.06 0"
                  transform="matrix(1 0 0 -1 0 29.744)"
                />
              </svg>
            </div>
            <div
              class="c5"
              style="opacity: 0;"
            >
              <svg
                height="15"
                opacity="1"
                viewBox="0 0 24 30"
                width="10"
              >
                <polygon
                  fill="#FFFFFF"
                  fill-rule="evenodd"
                  points="10.768 0 10.768 23.16 3.245 15.637 .776 18.106 12.414 29.744 24.052 18.106 21.583 15.637 14.06 23.16 14.06 0"
                  transform="translate(0 .256)"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </td>
  <td
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        title
        <div
          style="visibility: hidden;"
        >
          <div
            class="c4"
            data-testid="sort-indicators"
          >
            <div
              style="opacity: 0;"
            >
              <svg
                height="15"
                opacity="1"
                viewBox="0 0 24 30"
                width="10"
              >
                <polygon
                  fill="#FFFFFF"
                  fill-rule="evenodd"
                  points="10.768 0 10.768 23.16 3.245 15.637 .776 18.106 12.414 29.744 24.052 18.106 21.583 15.637 14.06 23.16 14.06 0"
                  transform="matrix(1 0 0 -1 0 29.744)"
                />
              </svg>
            </div>
            <div
              class="c5"
              style="opacity: 0;"
            >
              <svg
                height="15"
                opacity="1"
                viewBox="0 0 24 30"
                width="10"
              >
                <polygon
                  fill="#FFFFFF"
                  fill-rule="evenodd"
                  points="10.768 0 10.768 23.16 3.245 15.637 .776 18.106 12.414 29.744 24.052 18.106 21.583 15.637 14.06 23.16 14.06 0"
                  transform="translate(0 .256)"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </td>
</tr>
`;
